### PR TITLE
chore: upgraded electron to 14.1.0

### PR DESCRIPTION
**Pull request by vraravam**
#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has *stopped working* or is *working incorrectly*, please log the bug [here](https://github.com/getferdi/recipes/issues)
2. If you are requesting support for a **new service** in Ferdi, please log it [here](https://github.com/getferdi/recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/getferdi/ferdi#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdi, but to get new recipes between Ferdi releases, this documentation is quite useful.)
4. Please consider supporting Ferdi!
  👉  https://github.com/sponsors/getferdi
  👉  https://opencollective.com/getferdi/donate
5. Please ensure you've completed all of the following.
- [x] I have read the [Contributing Guidelines](https://github.com/getferdi/ferdi/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/getferdi/ferdi/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/getferdi/ferdi/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change
Upgrade `electron` to `14.1.0`

#### Motivation and Context
Upgrade path

#### Screenshots
Screenshot 2021-10-04 at 6 58 16 PM

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

**Comment by vraravam** [comment]
This is currently resulting in a white window without any content.

**Comment by mhatvan** [comment]
This PR also needs to include moving from using `electron` remote method to `@electron/remote` usage: https://www.electronjs.org/blog/electron-14-0#removed-remote-module

**Comment by vraravam** [comment]
> This PR also needs to include moving from using `electron` remote method to `@electron/remote` usage: https://www.electronjs.org/blog/electron-14-0#removed-remote-module

afaik - this was already done by @kris7t a couple of months ago when that separate namespace was introduced and we moved to electron v12.

**Comment by mhatvan** [comment]
I think it was reverted then due to some issues. I still see imports of remote from electron throughout the codebase.

**Comment by vraravam** [comment]
> I think it was reverted then due to some issues. I still see imports of remote from electron throughout the codebase.

im raising a PR for these - they can be merged independent of this PR.

**Comment by mhatvan** [comment]
> im raising a PR for these - they can be merged independent of this PR.

Yes, sounds good.
Can't find anymore that there were issues with replacing remote, might have imagined it.
All the leftover occurrences of deprecated remote module seem to be in /recipes, not ferdi code itself.
You can combine it with finishing up this issue: https://github.com/getferdi/ferdi/issues/1635

**Comment by mhatvan** [comment]
@vraravam Small heads up: electron 15 was released already so you might wanna change the version to latest.

**Comment by vraravam** [comment]
> @vraravam Small heads up: electron 15 was released already so you might wanna change the version to latest.

Yes - I know, but, i wanted to get this done with v14 so that any breaking changes are dealt with in an incremental fashion.

**Comment by vraravam** [comment]
New external dependency: `electron-react-titlebar` doesn't work
Logged [bug](https://github.com/KochiyaOcean/electron-react-titlebar/issues/21) for tracking purposes.

File: src/index.js [hunk 1]
```diff
@@ -39,11 +39,6 @@ import userAgent from './helpers/userAgent-helpers';
 
 const debug = require('debug')('Ferdi:App');
 
-// From Electron 9 onwards, app.allowRendererProcessReuse = true by default. This causes the app to crash on Windows due to the
-// Electron Windows Notification API crashing. Setting this to false fixes the issue until the electron team fixes the notification bug
-// More Info - https://github.com/electron/electron/issues/18397
-app.allowRendererProcessReuse = false;
```
[the remark this answers is not in the document]
Comment (vraravam) [reply src/index.js:49]: @kris7t - I am using the `twitter` service - and also running this latest code locally (attached the screenshot of the `About` dialog). I dont see crashes that you mentioned were in #1029

**Comment by vraravam** [comment]
im not getting the context menu to come up when running this code. Is it something that you can test @mhatvan / @kris7t ?

**Comment by vraravam** [comment]
UPDATE: I think something needs to change [here](https://github.com/vraravam/ferdi/blob/76a3e832057503ede72bfe5de633b0e2559833f7/src/components/services/content/ServiceWebview.js#L88) - just not able to figure out **what** though

**Comment by vraravam** [comment]
> UPDATE: I think something needs to change [here](https://github.com/vraravam/ferdi/blob/76a3e832057503ede72bfe5de633b0e2559833f7/src/components/services/content/ServiceWebview.js#L88) - just not able to figure out **what** though

Found the fix and all the above points are now fixed/working!!!

**Review by mhatvan — APPROVED** [review]

**Pull request merged**